mastermind.rb

```diff
@@ -0,0 +1,133 @@
+# Command Line Mastermind Game
+class Mastermind
+  attr_reader :player_score, :master_score, :player_guess,
+              :game_count, :board, :guess_count, :guess
+
+  def initialize
+    set_master_code
+    reset_board
+    reset_scores
+    reset_guess_count
+    display_rules
+    start_game
+  end
+
+  def reset_guess_count
+    @guess_count = 0
+  end
+
+  def set_master_code
+    @master_code = Array.new(4) { rand(2...8) }
+  end
+
+  def reset_scores
+    @player_score = 0
+    @master_score = 0
+  end
+
+  def reset_board
+    @board = Array.new(12) { Array.new(4) }
+    @win = false
+  end
+
+  def display_rules
+    print 'Welcome to Mastermind! A 4 letter code has been set by the mastermind. ' \
+          'You win if you can guess the 4 letter code in 12 turns. ' \
+          'After each guess, the mastermind will return the numbers 0 or 1. ' \
+          'Each "1" indicates that one of the guessed numbers is correct, but in the wrong place. ' \
+          'Each "2" indicates that one of the guessed numbers is correct and in the correct place. ' \
+          'You may guess any value between 2 - 7 inclusive. ' \
+          'Goodluck!!'
+  end
+
+  def update_game_count
+    @game_count.nil? ? @game_count = 1 : @game_count += 1
+  end
+
+  def check_guess_valid?(guess)
+    (guess.is_a? Array) && (guess.length == 4) && (guess.count { |e| e.between?(2,8) } == 4)
+  end
+
+  def start_game
+    update_game_count
+    if check_ready?
+      until game_over?
+        prompt_guess
+        if check_guess_valid?(@guess)
+          update_board
+          analyze_input
+          print_board
+        else
+          prompt_guess
+        end
+      end
+      determine_winner
+    end
+  end
+
+  def determine_winner
+    if @win
+      puts 'HOORAY! YOU CRACKED THE CODE'
+      puts @master_code
+      puts "in #{guess_count} tries!"
+      @player_score += 1
+    else
+      puts 'Mastermind has out master-minded you... :('
+      puts @master_code
+      puts '... Better luck next time!'
+      @master_score += 1
+    end
+    puts 'Player Score, Mastermind Score'
+    puts "#{player_score}, #{master_score}"
+  end
+
+  def game_over?
+    @win || @guess_count >= 12
+  end
+
+  def check_ready?
+    puts 'Are you ready to play? (Y/N)'
+    return true if gets.chomp == 'Y' || gets.chomp == 'yes' || gets.chomp == 'y'
```

Comment:
I'm thinking this won't work the way you think it'll work. Guessing you want something like ...
```
response = gets.chomp.downcase
return true if response == 'Y' || response == 'yes'
return false
```